### PR TITLE
Decompose metering point details

### PR DESCRIPTION
**Pull request by kft**
## Description
`MeteringPointDetails` used to be specific for consumption metering points. This PR decomposes into specific classes for `MeteringPoint`, `MarketMeteringPoint`, and `ConsumptionMeteringPoint`.

### CODE REVIEW

**Review by cmgrd — APPROVED** [review]

File: source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/MeteringPointDetails.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        MeteringPointId Id,
+        GsrnNumber GsrnNumber,
+        Address Address,
+        MeteringMethod MeteringMethod,
+        GridAreaLinkId GridAreaLinkId,
+        GsrnNumber? PowerPlantGsrnNumber,
+        LocationDescription? LocationDescription,
+        MeterId? MeterNumber,
+        ReadingOccurrence ReadingOccurrence,
+        PowerLimit PowerLimit,
```
Comment (erduna) [inline source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/MeteringPointDetails.cs:32]: this is optional
image

File: source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/MarketMeteringPoints/MeteringPointDetails.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        MeteringPointId Id,
+        GsrnNumber GsrnNumber,
+        Address Address,
+        MeteringMethod MeteringMethod,
+        GridAreaLinkId GridAreaLinkId,
+        GsrnNumber? PowerPlantGsrnNumber,
+        LocationDescription? LocationDescription,
+        MeterId? MeterNumber,
+        ReadingOccurrence ReadingOccurrence,
+        PowerLimit PowerLimit,
```
Comment (erduna) [inline source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/MarketMeteringPoints/MeteringPointDetails.cs:31]: This is optional
image

**Review by erduna — CHANGES_REQUESTED** [review]
Power limits (kwh, ampere) are optional fields. This also means the rules need to be changed so they don't expect them to be present.

File: source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/MeteringPointDetails.cs (hunk 1 above)
Comment (erduna) [reply source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/MeteringPointDetails.cs:32]: Will create issue on this not to block this pr, agreed with kft

**Review by erduna — APPROVED** [review]
Will create issue on power limit so as to not block this pr, agreed with kft

**Pull request merged**